graph_generation/generate_from_config.py

```diff
@@ -0,0 +1,230 @@
+        collection_count=config["collection"]["collection_count"]
+    )
+
+    processing_params = ProcessingParams(
+        dataset_criticality_proba_map=process_map(config["data_processing"]["dataset_criticality_proba_map"],
+                                                  proba=True, enum=True),
+        dataset_impact_proba_map=process_map(config["data_processing"]["dataset_impact_proba_map"], proba=True,
+                                             enum=True)
+    )
+
+    data_integrity_params = DataIntegrityParams(
+        data_restoration_range_seconds=config["data_integrity"]["restoration_range_seconds"],
+        data_regeneration_range_seconds=config["data_integrity"]["regeneration_range_seconds"],
+        data_reconstruction_range_seconds=config["data_integrity"]["reconstruction_range_seconds"],
+        data_volatility_proba_map=process_map(config["data_integrity"]["volatality_proba_map"], proba=True)
+    )
+
+    # Generate random connections between nodes.
+    graph_connections = ConnectionGenerator(
+        dataset_params=dataset_params,
+        system_params=system_params,
+        dataset_to_system_params=dataset_to_system_params,
+        collection_params=collection_params
+    )
+    graph_connections.generate()
+    logging.info(f"Successfully generated connections.")
+
+    # Number of connections can not be predicted, so we need to get them for attribute generation.
+    system_write_conn_count = sum([len(i) for i in graph_connections.dataset_write_conn_systems.values()])
+    system_read_conn_count = sum([len(i) for i in graph_connections.dataset_read_conn_systems.values()])
+    connection_params = ConnectionParams(system_write_conn_count, system_read_conn_count)
+
+    # Generate attributes.
+    graph_attributes = AttributeGenerator(dataset_params, system_params, data_integrity_params, processing_params,
+                                          connection_params)
+    graph_attributes.generate()
+    logging.info(f"Successfully generated attributes.")
+
+    # Create a graph.
+    if graph_type == "proto":
+        graph = GraphTemplate().get_proto_graph()
+    else:
+        graph = GraphTemplate().get_nx_graph()
+    logging.info(f"Created empty {graph_type} graph.")
+
+    # Generate collections.
+    start = time.time()
+    for collection_id in range(1, collection_params.collection_count + 1):
+        graph.generate_collection(collection_id)
+    logging.info(f"Generated collections in {round(time.time() - start, 1)} seconds.")
+
+    # Generate dataset collections.
+    start = time.time()
+    for collection_id in graph_connections.dataset_collections_conn_collection:
+        for dataset_collection_id in graph_connections.dataset_collections_conn_collection[collection_id]:
+            graph.generate_dataset_collection(dataset_collection_id, collection_id)
+    logging.info(f"Generate dataset collections in {round(time.time() - start, 1)} seconds.")
```

Comment:
nit: generated   (here and elsewhere)
 ______________^

Also shouldn't this be "after x seconds" rather than "in x seconds"?

Reply:
Will change to generated. I think it should be "in x seconds" (as comment describes the amount of time needed to do something)

Reply:
You're right, I misread things